# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/kasc/crimson/crimson_vidinit.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- /home/fizz/dev/games/crimson/tools/cfed.cpp	Fri Apr 16 17:56:33 2004
+++ ./tools/cfed.cpp	Mon May 17 18:57:12 2004
@@ -105,7 +105,7 @@
   char *uset = NULL, *tset = NULL;
   string outname;
 
-  if ( SDL_Init( SDL_INIT_VIDEO ) < 0 ) {
+  if ( SDL_Init( 0 ) < 0 ) {
     cerr << "Error: Could not init SDL. " << SDL_GetError() << endl;
     return 1;
   }
--- /home/fizz/dev/games/crimson/src/common/surface.cpp	Fri Feb 27 19:54:12 2004
+++ ./src/common/surface.cpp	Mon May 17 18:59:50 2004
@@ -91,7 +91,7 @@
 int Surface::DisplayFormat( void ) {
   SDL_Surface *new_s;
 
-  if ( s_surface == 0 ) return -1;
+  if ( s_surface == 0 || !SDL_WasInit( SDL_INIT_VIDEO ) )  return -1;
   if ( (new_s = SDL_DisplayFormat( s_surface )) == 0 ) return -1;
   SDL_FreeSurface( s_surface );
   s_surface = new_s;
